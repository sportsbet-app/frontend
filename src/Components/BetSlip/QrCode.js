import React from 'react'
import QR from 'qrcode.react'
export default function QrCode(props) {
    const { bet } = props
    const value = (JSON.stringify(bet))
    return (
        <div>
            <QR value={bet && value} renderAs={'svg'} level={'L'} />
        </div>

    )

}




