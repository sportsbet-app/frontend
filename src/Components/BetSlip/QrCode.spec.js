import React from 'react'
//import all dependencies needed
import { render, act } from '@testing-library/react'
import { getElement, shallow } from 'enzyme'
//import renderer for snapshot
import renderer from 'react-test-renderer'
//import the component being tested
import QrCode from './QrCode'

//test data to use for the render
const data = { one: 'http://tester.facebook.com' }

//test to check that the app will display when called.
describe('<QrCode />', () => {
  it('should display', () => {
    //check the render with the test data 
    render(<QrCode bet={data} />)

  })

  //create a snapshot
  it('matches snapshot', () => {
    const tree = renderer.create(<QrCode bet={data} />); // generates a DOM tree

    // snapshots are a JSON representation of the DOM tree
    expect(tree.toJSON()).toMatchSnapshot();
  });

  //check that a qr svg image is created
  it('must display an svg', () => {
    //create render
    const { getByTestId, findByTestId } = render(<QrCode bet={data} />)
    //create variable to be the instance of the container after rendering
    const div = getByTestId('container')
    const page = findByTestId('container')

    act(() => {
      div.getElement('svg')
    })
    //create a variable for the QR code
    const QR = document.getElementsByTagName('*')
    console.log(QR, 'div', div)
    //and check it to be true
    expect(QR)
  })
})
