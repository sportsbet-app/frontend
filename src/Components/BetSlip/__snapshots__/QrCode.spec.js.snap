// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QrCode /> matches snapshot 1`] = `
<div
  data-testid="container"
>
  <svg
    height={128}
    shapeRendering="crispEdges"
    viewBox="0 0 29 29"
    width={128}
  >
    <path
      d="M0,0 h29v29H0z"
      fill="#FFFFFF"
    />
    <path
      d="M0 0h7v1H0zM8 0h3v1H8zM12 0h1v1H12zM14 0h1v1H14zM18 0h1v1H18zM22,0 h7v1H22zM0 1h1v1H0zM6 1h1v1H6zM8 1h3v1H8zM16 1h2v1H16zM19 1h2v1H19zM22 1h1v1H22zM28,1 h1v1H28zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM8 2h1v1H8zM10 2h2v1H10zM13 2h1v1H13zM15 2h1v1H15zM17 2h1v1H17zM19 2h2v1H19zM22 2h1v1H22zM24 2h3v1H24zM28,2 h1v1H28zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM8 3h2v1H8zM11 3h1v1H11zM14 3h1v1H14zM18 3h2v1H18zM22 3h1v1H22zM24 3h3v1H24zM28,3 h1v1H28zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM13 4h4v1H13zM18 4h1v1H18zM20 4h1v1H20zM22 4h1v1H22zM24 4h3v1H24zM28,4 h1v1H28zM0 5h1v1H0zM6 5h1v1H6zM8 5h1v1H8zM11 5h4v1H11zM18 5h3v1H18zM22 5h1v1H22zM28,5 h1v1H28zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14 6h1v1H14zM16 6h1v1H16zM18 6h1v1H18zM20 6h1v1H20zM22,6 h7v1H22zM9 7h1v1H9zM12 7h1v1H12zM14 7h1v1H14zM16 7h4v1H16zM0 8h2v1H0zM4 8h3v1H4zM11 8h1v1H11zM13 8h2v1H13zM17 8h1v1H17zM19 8h1v1H19zM23 8h1v1H23zM25,8 h4v1H25zM0 9h1v1H0zM2 9h1v1H2zM4 9h2v1H4zM8 9h3v1H8zM12 9h1v1H12zM15 9h1v1H15zM18 9h1v1H18zM21 9h2v1H21zM24 9h1v1H24zM28,9 h1v1H28zM2 10h1v1H2zM4 10h4v1H4zM11 10h5v1H11zM20 10h2v1H20zM23 10h1v1H23zM26 10h1v1H26zM28,10 h1v1H28zM0 11h1v1H0zM2 11h3v1H2zM8 11h1v1H8zM10 11h2v1H10zM13 11h1v1H13zM15 11h1v1H15zM17 11h2v1H17zM20 11h2v1H20zM23 11h1v1H23zM25 11h1v1H25zM1 12h2v1H1zM5 12h3v1H5zM10 12h1v1H10zM12 12h2v1H12zM15 12h1v1H15zM17 12h1v1H17zM19 12h2v1H19zM23 12h1v1H23zM25 12h1v1H25zM27,12 h2v1H27zM3 13h2v1H3zM8 13h1v1H8zM13 13h4v1H13zM18 13h1v1H18zM22 13h3v1H22zM26 13h1v1H26zM28,13 h1v1H28zM0 14h3v1H0zM6 14h1v1H6zM8 14h2v1H8zM15 14h1v1H15zM17 14h2v1H17zM23 14h1v1H23zM25 14h1v1H25zM28,14 h1v1H28zM0 15h6v1H0zM7 15h1v1H7zM9 15h2v1H9zM12 15h1v1H12zM14 15h1v1H14zM17 15h2v1H17zM20 15h1v1H20zM27 15h1v1H27zM0 16h1v1H0zM2 16h1v1H2zM6 16h4v1H6zM11 16h1v1H11zM13 16h3v1H13zM17 16h1v1H17zM19 16h1v1H19zM23 16h1v1H23zM25 16h1v1H25zM0 17h5v1H0zM9 17h1v1H9zM12 17h1v1H12zM15 17h2v1H15zM18 17h1v1H18zM22 17h5v1H22zM28,17 h1v1H28zM3 18h13v1H3zM20 18h1v1H20zM23 18h1v1H23zM26,18 h3v1H26zM2 19h1v1H2zM4 19h2v1H4zM7 19h2v1H7zM10 19h2v1H10zM13 19h1v1H13zM16 19h2v1H16zM19 19h1v1H19zM0 20h2v1H0zM3 20h6v1H3zM10 20h1v1H10zM12 20h2v1H12zM17 20h1v1H17zM19 20h6v1H19zM28,20 h1v1H28zM8 21h1v1H8zM13 21h6v1H13zM20 21h1v1H20zM24 21h2v1H24zM27,21 h2v1H27zM0 22h7v1H0zM10 22h1v1H10zM15 22h2v1H15zM18 22h3v1H18zM22 22h1v1H22zM24 22h1v1H24zM26 22h1v1H26zM28,22 h1v1H28zM0 23h1v1H0zM6 23h1v1H6zM8 23h2v1H8zM12 23h1v1H12zM14 23h1v1H14zM17 23h4v1H17zM24 23h1v1H24zM27,23 h2v1H27zM0 24h1v1H0zM2 24h3v1H2zM6 24h1v1H6zM8 24h4v1H8zM13 24h3v1H13zM17 24h1v1H17zM19 24h7v1H19zM27 24h1v1H27zM0 25h1v1H0zM2 25h3v1H2zM6 25h1v1H6zM9 25h1v1H9zM12 25h1v1H12zM15 25h1v1H15zM17 25h2v1H17zM20 25h2v1H20zM23 25h1v1H23zM25 25h2v1H25zM28,25 h1v1H28zM0 26h1v1H0zM2 26h3v1H2zM6 26h1v1H6zM9 26h1v1H9zM11 26h5v1H11zM18 26h2v1H18zM26,26 h3v1H26zM0 27h1v1H0zM6 27h1v1H6zM8 27h2v1H8zM11 27h1v1H11zM13 27h1v1H13zM16 27h3v1H16zM20 27h1v1H20zM22 27h2v1H22zM25 27h1v1H25zM27,27 h2v1H27zM0 28h7v1H0zM8 28h3v1H8zM12 28h2v1H12zM15 28h1v1H15zM17 28h1v1H17zM20 28h1v1H20zM25 28h1v1H25zM27 28h1v1H27z"
      fill="#000000"
    />
  </svg>
</div>
`;
